WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -587,8 +778,34 @@ extension PublishingEditor {
         debouncer.call(immediate: immediate)
     }
 
-    // TODO: Rip this out and put it into the PostService
     func createRevisionOfPost(loadAutosaveRevision: Bool = false) {
+        guard RemoteFeatureFlag.syncPublishing.enabled() else {
+            return _createRevisionOfPost(loadAutosaveRevision: loadAutosaveRevision)
+        }
+        guard let managedObjectContext = post.managedObjectContext else {
+            return assertionFailure()
+        }
+
+        assert(post.latest() == post, "Must be opened with the latest verison of the post")
+
+        if !(post.isRevision() && !post.isSyncNeeded) {
```

Comment:
This check is needed only for a scenario where the app crashes during editor – it will be covered later. I plan to re-open the editor automatically to make sure the user either commits of discard the revision.